src/uct/ib/rc/base/rc_ep.c

```diff
@@ -488,6 +489,26 @@ ucs_status_t uct_rc_ep_flush(uct_rc_ep_t *ep, int16_t max_available,
     return UCS_INPROGRESS;
 }
 
+ucs_status_t uct_rc_ep_flush_cancel(uct_rc_ep_t* ep)
+{
+    uct_rc_iface_t *iface      = ucs_derived_of(ep->super.super.iface,
+                                                uct_rc_iface_t);
+    uct_rc_iface_ops_t *rc_ops = ucs_derived_of(iface->super.ops,
+                                                uct_rc_iface_ops_t);
+    ucs_status_t status;
+
+    ucs_assert(!(ep->flags & UCT_RC_EP_FLAG_FLUSH_CANCEL));
+    status = rc_ops->fail_ep(&ep->super.super);
```

Comment:
can we avoid adding "virtual" function: the flush_ep(cancel) code will call modigy_qp, and then common function to set fc_wnd and flags